pkg/storaged/format-dialog.jsx

```diff
@@ -356,8 +356,8 @@ export function format_dialog(client, path, start, size, enable_dos_extended) {
                 return utils.teardown_active_usage(client, usage)
                         .then(utils.reload_systemd)
                         .then(format)
-                        .then(maybe_mount)
-                        .then(utils.reload_systemd);
+                        .then(utils.reload_systemd)
+                        .then(maybe_mount);
```

Comment:
Is this synchronous enough? I was fairly sure that Reload() was async. You may need to wait for the `Reloading(False)` signal here?

Reply:
It certainly slows down everything enormously... I'll check to make sure.

Reply:
The "method_return" for the call to Reload comes after the signal that informs about "Reloading=false":

```
‣ Type=method_call  Endian=l  Flags=4  Version=1 Cookie=2  Timestamp="Mon 2021-05-24 12:02:28.097747 UTC"
  Sender=:1.40  Destination=org.freedesktop.systemd1  Path=/org/freedesktop/systemd1  Interface=org.freedesktop.systemd1.Manager  Member=Reload
  UniqueName=:1.40
  MESSAGE "" {
  };
[...]
‣ Type=signal  Endian=l  Flags=1  Version=1 Cookie=7132  Timestamp="Mon 2021-05-24 12:02:28.650280 UTC"
  Sender=:1.2  Path=/org/freedesktop/systemd1  Interface=org.freedesktop.systemd1.Manager  Member=Reloading
  UniqueName=:1.2
  MESSAGE "b" {
          BOOLEAN false;
  };

‣ Type=method_return  Endian=l  Flags=1  Version=1 Cookie=7133  ReplyCookie=2  Timestamp="Mon 2021-05-24 12:02:28.650307 UTC"
  Sender=:1.2  Destination=:1.40
  UniqueName=:1.2
  MESSAGE "" {
  };
```

That's enough proof for me.

Reply:
Thanks for checking! The "slows down enormously" is both good and bad, I figure :grin: 

Good to know that Reload() is synchronous, that's indeed much more helpful for API users.

Reply:
> The "slows down enormously" is both good and bad, I figure.

It's not so bad during formatting (although we should say in the dialog that it is happeneing, "Reloading system configuration", maybe).  It is a bit annoying during unmounting, which would otherwise be instant.  Also, it doesn't help that we need to do it twice...